File: iv.py
#IV Calculator for Pokemon GO 
# Value range for A/D/S --> 0 - 15.

import math
def rounded(value):
    dec_val = value - int(value)
    first_dec_digit = int(dec_val * 10)
    if 0 <= first_dec_digit <= 4:
        return int(value)
    else:
        return int(value)+1

try:
    A=int(input("\nEnter Attack stat: "))
    D=int(input("\nEnter Defense stat: "))
    S=int(input("\nEnter Stamina stat: "))

    if not all(0 <= stat <= 15 for stat in (A, D, S)):
        print("\nOne or more values are out of range.")
        exit
    else:
        total = A + D + S
        result = total * (20/9)    
        res_rounded = rounded(float(f"{result:.2f}"))
        print(f"\nIV = {res_rounded}%.")
except ValueError:
    print("\nInvalid Input. Program terminated.")
    exit

File: iv.c
// IV calculator for Pokemon GO.
// All stats range from 0-15

#include <stdio.h>
#include <math.h>
#include<stdlib.h>
int result(float total){
	float diff= total - (int)total;
	int first_digit=(int)(10*diff);
	if((first_digit>=0) && (first_digit<=4))
	return (int)total;
	else
	return 1 + (int)total;
}
int main(){
	int A=0,D=0,S=0;
	float total=0.0;
	
	while(1){
	printf("\nEnter Attack stat: ");
	scanf("%d" ,&A);
	printf("\nEnter Defense stat: ");
	scanf("%d" ,&D);
	printf("\nEnter Stamina stat: ");
	scanf("%d" ,&S);
	
	if((A>=0 && A<=15) && (D>=0 && D<=15) && (S>=0 && S<= 15))
	break;
	else{
	printf("\nOne or more values exceed the range.");
	exit(1);
}
	}
	
	total = (A+D+S)* 20.0/9.0;
	total = roundf(total*100)/100;
	printf("\nIV = %d." ,result(total));
	return 0;
}
